lib/widgets/routers/splash_router.dart

```diff
@@ -1,31 +1,44 @@
 import 'package:coffeecard/cubits/authentication/authentication_cubit.dart';
 import 'package:coffeecard/cubits/environment/environment_cubit.dart';
 import 'package:coffeecard/widgets/pages/home_page.dart';
-import 'package:coffeecard/widgets/pages/splash/splash_error_page.dart';
-import 'package:coffeecard/widgets/pages/splash/splash_loading_page.dart';
-import 'package:coffeecard/widgets/routers/entry_router.dart';
+import 'package:coffeecard/widgets/routers/login_router.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 
 class SplashRouter extends StatefulWidget {
-  const SplashRouter();
+  const SplashRouter({required this.navigatorKey, required this.child});
+
+  final GlobalKey<NavigatorState> navigatorKey;
+  final Widget child;
 
   @override
   _SplashRouterState createState() => _SplashRouterState();
 }
 
 class _SplashRouterState extends State<SplashRouter> {
-  /// Navigates out of the splash screen if both
-  /// authentication status and enviornment status are loaded.
-  void _maybeNavigate(BuildContext context) {
-    final envState = context.read<EnvironmentCubit>().state;
-    final authStatus = context.read<AuthenticationCubit>().state.status;
+  var _authStatus = AuthenticationStatus.unknown;
+  bool? _isTestEnvironment;
+
+  void _authListener(BuildContext _, AuthenticationState state) {
+    _authStatus = state.status;
+    _maybeNavigate();
+  }
+
+  void _envListener(BuildContext _, EnvironmentState state) {
+    if (state is EnvironmentError) {
+      // FIXME: Handle error
+    }
+    _isTestEnvironment = state is EnvironmentLoaded && state.isTestEnvironment;
```

Comment:
This assignment is never null, so the later check for whether the environment is null, will always be false